apps/veil/src/pages/tournament/server/delegator-history.ts

```diff
@@ -32,162 +31,118 @@ export interface TournamentDelegatorHistoryResponse {
-  return pindexerDb
-    .selectFrom('lqt.delegator_history')
-    .orderBy('epoch', 'desc')
-    .orderBy('power', 'desc')
-    .where('epoch', '=', epoch)
-    .selectAll()
-    .execute();
-};
+export const getBodyParams = async (
+  req: NextRequest,
+): Promise<Required<DelegatorHistoryRequest> | string> => {
+  const body = (await req.json()) as DelegatorHistoryRequest;
 
-/**
- * Consolidate per-epoch query results into a single history for the given
- * delegator address.
- *
- * For each epoch we:
- *    * filter rows belonging to the delegator,
- *    * tally their total voting power,
- *    * capture that epoch’s reward and asset id.
- *
- * Finally, we aggregate the rewards earned by the address across all epochs.
- */
-function processEpochResults(
-  epochResults: {
-    epoch: string;
-    data: { address: Buffer; epoch: number; power: number; asset_id: Buffer; reward: number }[];
-  }[],
-  targetAddress: string,
-  address: Address,
-  limit: number,
-  page: number,
-  sortDirection: DelegatorHistorySortDirection = 'desc',
-): { paginatedResults: LqtDelegatorHistory[]; totalItems: number; totalReward: number } {
-  // Create accumulator for results
-  const matchesByEpoch = new Map<string, { power: number; reward: number; assetId: Buffer }>();
-
-  // Create result entries and calculate total reward.
-  const results: LqtDelegatorHistory[] = [];
-  let totalReward = 0;
-
-  for (const { epoch, data } of epochResults) {
-    // Find all matching items for this epoch.
-    const matches = data.filter(
-      item => bech32mAddress(new Address({ inner: item.address })) === targetAddress,
-    );
+  const limit = Number(body.limit) || BASE_LIMIT;
+  const page = Number(body.page) || BASE_PAGE;
 
-    if (matches.length > 0 && matches[0]) {
-      // Calculate total power for matches, and store first match's reward and asset_id.
-      const totalPower = matches.reduce((sum, item) => sum + Number(item.power), 0);
-      matchesByEpoch.set(epoch, {
-        power: totalPower,
-        reward: Number(matches[0].reward),
-        assetId: matches[0].asset_id,
-      });
-    }
-  }
+  const sortKeyParam = body.sortKey;
+  const sortKey = sortKeyParam && SORT_KEYS.includes(sortKeyParam) ? sortKeyParam : 'epoch';
 
-  for (const [epoch, match] of matchesByEpoch.entries()) {
-    results.push({
-      address: Buffer.from(address.inner),
-      epoch: parseInt(epoch, 10),
-      power: match.power,
-      asset_id: Buffer.from(match.assetId),
-      reward: match.reward,
-    });
+  const sortDirectionParam = body.sortDirection;
+  const sortDirection =
+    sortDirectionParam && DIRECTIONS.includes(sortDirectionParam) ? sortDirectionParam : 'desc';
 
-    totalReward += match.reward;
+  const epochs = Array.isArray(body.epochs)
+    ? body.epochs.filter(epoch => typeof epoch === 'string')
+    : [];
+
+  if (!isAddress(body.address)) {
+    return 'Invalid address';
   }
```

Comment:
Instead of doing a check like this, we should instead transform the type so that it's an actual Address type. This isn't blocking though, but a point to think of when we refactor the domain type library and what not.